crates/sui-framework-build/src/unit_tests/build_tests.rs

```diff
@@ -22,3 +25,41 @@ fn generate_struct_layouts() {
         registry.contains_key("0000000000000000000000000000000000000002::tx_context::TxContext")
     );
 }
+
+#[test]
+fn generate_lock_file() {
```

Comment:
I put this test in `sui-framework-build`, not sure if it's the best place. I considered `cli-tests` and `sui-core/src/unit_tests/move_integration_tests.rs` but these didn't look like a great fit either.